src/others/file_explorer/lv_file_explorer.c

```diff
@@ -522,10 +535,17 @@ static void browser_file_event_handler(lv_event_t * e)
             else { /* Nothing to do*/ }
         }
 
-        lv_fs_dir_t dir;
-        if(lv_fs_dir_open(&dir, file_name) == LV_FS_RES_OK) {
-            lv_fs_dir_close(&dir);
-            show_dir(obj, (char *)file_name);
+        /* Navigate to the file path if this is a directory */
+        if(file_entry_user_data->file_kind == LV_FILE_EXPLORER_FILE_KIND_DIR) {
+            lv_fs_dir_t dir;
+            lv_fs_res_t res = lv_fs_dir_open(&dir, file_name);
+            if(res == LV_FS_RES_OK) {
+                lv_fs_dir_close(&dir);
+                show_dir(obj, (char *)file_name);
```

Comment:
I was testing and found that I was relying on some unreliable behavior. The filesystem drivers currently cannot tell if a symbolic link is a directory, so the icon becomes `FILE_KIND_FILE` but it was ok because the highlighted logic would always try to open the entry as a dir first, and if it fails, it's a file, so I could still open symlink directories. But after this change, it can no longer. I think I may open a PR for `lv_fs_posix` to recognize symlink directories as directories.

Reply:
I think that it is a good idea to fix this on the fs driver level